bindings/java/nz/mega/sdk/MegaChatApiJava.java

```diff
@@ -1750,6 +1750,8 @@ public void closeChatRoom(long chatid, MegaChatRoomListenerInterface listener){
             DelegateMegaChatRoomListener item = itr.next();
             if(item.getUserListener() == listener){
                 listenerToDelete = item;
+                boolean success = listenerToDelete.invalidateUserListener();
+                assert success; // failed if listener was already invalidated
```

Comment:
Assertions are unreliable on Android. `itr.remove();` will execute even when failed, is it expected?

Reply:
Assertions are unreliable?? what does it mean?
Assertions are conditions that are only checked in debugging mode. Every developer must code using debugging mode, so any failed assertion stops the execution and you can check why it failed.
However, in release mode (like it runs in production, regardless logs are enabled or not), the assertion will not be checked at all and therefore the `itr.remove()` will execute.

Reply:
What do you mean by _Assertions are unreliable_?

Assertions are conditions that are only checked in debugging mode. Every developer should usually code using debugging mode, so any failed assertion stops the execution and the developer can check why it failed.
However, in release mode (like it runs in production, regardless logs are enabled or not), the assertion will not be checked at all and therefore the `itr.remove()` will execute, as it should be :)

Reply:
Assertions are unreliable in Dalvik and unimplemented in ART. Should use `BuildConfig.DEBUG` instead.
But only for debugging, it doesn't matter.